### PR TITLE
add missing tests from meet embedders testing plan

### LINKED ISSUE #38556: Post RC1 refactors for Meet embedders
```[tasklist]
### Tasks
- [ ] https://github.com/metabase/metabase/pull/38669
- [ ] https://github.com/metabase/metabase/pull/38708
- [x] make sure the snowplow e2e tests all the events (specifically: at least app the steps)
- [ ] https://github.com/metabase/metabase/pull/38772
```

### PR DESCRIPTION
**Pull request by npretto**
Last bits of https://github.com/metabase/metabase/issues/38556
Closes https://github.com/metabase/metabase/issues/38556

### Description

Should add the missing tests from [Testing plan for v49](https://www.notion.so/metabase/Testing-plan-for-v49-d0e68bb6c94f4ff7b37860cd5b7faa35)

### CODE REVIEW

File: frontend/src/metabase/home/components/HomeContent/HomeContent.unit.spec.tsx [hunk 1]
```diff
@@ -173,4 +180,64 @@ describe("HomeContent", () => {
...
+      screen.getByRole("button", { name: "close icon" }).click();
+
+      expect(
+        screen.queryByText("Get started with Embedding Metabase in your app"),
+      ).not.toBeInTheDocument();
+
+      expect(localStorage.getItem("showEmbedHomepage")).toBeNull();
+    });
+
+    it("should now show it if the user is not admin", async () => {
```
Comment (WiNloSt) [inline frontend/src/metabase/home/components/HomeContent/HomeContent.unit.spec.tsx:222]:
```suggestion
    it("should not show it if the user is not admin", async () => {
```

File: frontend/src/metabase/home/components/HomeContent/HomeContent.unit.spec.tsx [hunk 2]
```diff
@@ -173,4 +180,64 @@ describe("HomeContent", () => {
...
+        user: createMockUser({ is_superuser: false }),
+        hasEmbeddingHomepageFlag: true,
+      });
+
+      expect(
+        screen.queryByText("Get started with Embedding Metabase in your app"),
+      ).not.toBeInTheDocument();
+    });
+
+    it("should now show it if the localStorage flag is not set", async () => {
```
Comment (WiNloSt) [inline frontend/src/metabase/home/components/HomeContent/HomeContent.unit.spec.tsx:233]:
```suggestion
    it("should not show it if the localStorage flag is not set", async () => {
```

**Review by WiNloSt — APPROVED** [review]
Looks good to me 👍

**Comment by WiNloSt** [comment]
Also I remember at one point we talked about `localStorage` not working in test. Isn't that why we need to have try...cach blocks in `frontend/src/metabase/home/utils.ts`?

I've tested that removing all of them still doesn't fail the unit tests. Would it be possible to remove them now?

**Comment by npretto** [comment]
> Also I remember at one point we talked about `localStorage` not working in test. Isn't that why we need to have try...cach blocks in `frontend/src/metabase/home/utils.ts`?
> 
> I've tested that removing all of them still doesn't fail the unit tests. Would it be possible to remove them now?

Some browsers block localStorage, for example we have this issue open for chrome in incognito: https://github.com/metabase/metabase/issues/35040

**Pull request merged**